File: app/src/main/java/pl/pola_app/internal/di/PolaComponent.java
package pl.pola_app.internal.di;

import javax.inject.Singleton;

import dagger.Component;
import pl.pola_app.PolaApplication;
import pl.pola_app.ui.activity.MainActivity;
import pl.pola_app.ui.fragment.DetailsFragment;
import pl.pola_app.ui.fragment.ProductDetailsFragment;
import pl.pola_app.ui.fragment.ScannerFragment;

@Singleton
@Component(modules = {OttoModule.class, WidgetModule.class, SharedPrefsModule.class})
public interface PolaComponent {


    final class Initializer {

        public static PolaComponent init(PolaApplication app) {
            return DaggerPolaComponent.builder()
                    .widgetModule(new WidgetModule(app))
                    .sharedPrefsModule(new SharedPrefsModule(app))
                    .build();
        }

    }

    void inject(MainActivity mainActivity);

    void inject(ScannerFragment scannerFragment);

    void inject(DetailsFragment detailsFragment);

    void inject(ProductDetailsFragment productDetailsFragment);
}

File: app/src/main/java/pl/pola_app/internal/di/PolaComponent.kt
package pl.pola_app.internal.di

import dagger.Component
import pl.pola_app.PolaApplication
import pl.pola_app.ui.activity.MainActivity
import pl.pola_app.ui.fragment.ProductDetailsFragment
import pl.pola_app.ui.fragment.ScannerFragment
import javax.inject.Singleton

@Singleton
@Component(modules = [OttoModule::class, WidgetModule::class, SharedPrefsModule::class])
interface PolaComponent {
    object Initializer {
        @JvmStatic
        fun init(app: PolaApplication): PolaComponent {
            return DaggerPolaComponent.builder()
                .widgetModule(WidgetModule(app))
                .sharedPrefsModule(SharedPrefsModule(app))
                .build()
        }
    }

    fun inject(mainActivity: MainActivity?)
    fun inject(scannerFragment: ScannerFragment)
    fun inject(productDetailsFragment: ProductDetailsFragment)
}